sfxclient/sfxclient.go

```diff
@@ -283,6 +271,81 @@ func (s *Scheduler) RemoveGroupedCallback(group string, db Collector) {
 	}
 }
 
+// CollectorAggregator is an
+type CollectorAggregator interface {
+	Var() expvar.Var
+	CollectDatapoints() []*datapoint.Datapoint
+	AddCallback(db Collector)
+	DefaultDimensions(dims map[string]string)
+	GroupedDefaultDimensions(group string, dims map[string]string)
+	AddGroupedCallback(group string, db Collector)
+	RemoveCallback(db Collector)
+	RemoveGroupedCallback(group string, db Collector)
+}
+
+// DefaultHTTPErrorHandler is the default way to handle errors by a CollectorHandler.  It attempts to reply with a 500 http error
+var DefaultHTTPErrorHandler = func(err error, w http.ResponseWriter, req *http.Request) error {
+	http.Error(w, err.Error(), http.StatusInternalServerError)
+	return err
+}
+
+// CollectorHandler aggregates collectors handles http requests for datapoints
+type CollectorHandler struct {
+	*CollectorCallbackManager
+	HTTPErrorHandler   func(error, http.ResponseWriter, *http.Request) error
+	jsonMarshallerFunc func(v interface{}) ([]byte, error)
```

Comment:
this method doesn't really have to be a json function tho does it? just a function that turns an object into bytes? could be protobuf, or something else...

Reply:
it just strikes me that this file now has a json library in it... which seems like something that it doesn't need or want.

Reply:
Fair point, it could be anything.  I can generalize it more.  We probably do need the json import and a default Marshaller function so we can return something by default.